### PR TITLE
CLOUDP-78065: [mongocli] atlas logs download doesn’t work if --out is not provided

### PR DESCRIPTION
**Pull request by gssbzn**
## Proposed changes

_Jira ticket:_ CLOUDP-78065

## Checklist

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

File: internal/cli/atlas/logs/download.go [hunk 1]
```diff
@@ -65,10 +65,11 @@ func (opts *DownloadOpts) Run() error {
 	return f.Close()
 }
 
-func (opts *DownloadOpts) initDefaultOut() {
+func (opts *DownloadOpts) initDefaultOut() error {
 	if opts.Out == "" {
 		opts.Out = strings.ReplaceAll(opts.name, ".gz", ".log.gz")
 	}
+	return nil
 }
```
Comment (andreaangiolillo) [inline internal/cli/atlas/logs/download.go:73]: [q] in which scenario this function would return an error?

**Review by andreaangiolillo — COMMENTED** [review]
I have one question but in general, I don't see which was the cause of the error

File: internal/cli/atlas/logs/download.go (hunk 1 above)
Comment (gssbzn) [reply internal/cli/atlas/logs/download.go:73]: never, this is just so this function can now be used in
```go
opts.PreRunE(opts.ValidateProjectID, opts.initStore, opts.initDefaultOut)
```

File: internal/cli/atlas/logs/download.go [hunk 2]
```diff
@@ -89,12 +90,11 @@ func DownloadBuilder() *cobra.Command {
...
 		PreRunE: func(cmd *cobra.Command, args []string) error {
-			opts.initDefaultOut()
-			return opts.PreRunE(opts.ValidateProjectID, opts.initStore)
-		},
-		RunE: func(cmd *cobra.Command, args []string) error {
 			opts.host = args[0]
 			opts.name = args[1]
+			return opts.PreRunE(opts.ValidateProjectID, opts.initStore, opts.initDefaultOut)
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
```
Comment (gssbzn) [inline internal/cli/atlas/logs/download.go:97]: @andreaangiolillo the error is here, I was calling `initDefaultOut` before `opts.name = args[1]` had any value so it was being overwritten by the Run command

Comment (andreaangiolillo) [reply internal/cli/atlas/logs/download.go:97]: oh got it, thanks for the clarification

**Review by andreaangiolillo — APPROVED** [review]
LGTM

**Pull request merged**